tests/integ/test_df_to_snowpark_pandas.py

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+# Tests behavior of to_snowpark_pandas() without explicitly initializing Snowpark pandas.
+
+import pytest
+
+from snowflake.snowpark._internal.utils import TempObjectType
+from tests.utils import Utils
+
+
+@pytest.fixture(scope="module")
+def tmp_table_basic(session):
+    table_name = Utils.random_name_for_temp_object(TempObjectType.TABLE)
+    Utils.create_table(
+        session, table_name, "id integer, foot_size float, shoe_model varchar"
+    )
+    session.sql(f"insert into {table_name} values (1, 32.0, 'medium')").collect()
+    session.sql(f"insert into {table_name} values (2, 27.0, 'small')").collect()
+    session.sql(f"insert into {table_name} values (3, 40.0, 'large')").collect()
+
+    try:
+        yield table_name
+    finally:
+        Utils.drop_table(session, table_name)
+
+
+def test_to_snowpark_pandas_no_modin(session, tmp_table_basic):
+    snowpark_df = session.table(tmp_table_basic)
+    # Check if modin is installed
```

Comment:
i am little bit confused here about the behavior, what happens under the following scenario before and after your change:
1: modin is not installed
2: modin is installed but user didn't do" import modin.pandas as pd; import snowflake.snowpark.modin.plugin"

i assume now under case 2, once we call snowpark_df.to_snowpark_pandas(), user will be able to perform snowpark pandas dataframe operations on top it just fine. what happens if user do the following after that?
```
import modin.pandas as pd
pd.read_snowflake(xx) 
```

Reply:
Yes, in case 2 any operations on `modin.pandas` would be with Snowpark pandas.

Reply:
Can you give me a brief description about the behavior before and after your change for the above 2 scenarios ?

Reply:
**Previous (release 1.18, any commit prior to 40a534e)**
1. modin not installed: Calling `to_snowpark_pandas_dataframe` raises `ModuleNotFoundError: Modin is not installed. Run `pip install "snowflake-snowpark-python[modin]"` to resolve.`
2. modin is installed: After calling `to_snowpark_pandas_dataframe`, the Snowpark pandas plugin is initialized and after running `import modin.pandas as pd`, users can perform Snowpark pandas operations as normal.

**Current (releases 1.19/1.20)**
1. modin not installed: Calling `to_snowpark_pandas_dataframe` raises `ModuleNotFoundError: No module named 'modin'. The error message is different because it bypasses the Snowpark pandas plugin initialization code which would validate the modin version, and instead directly runs code that tries to import Modin.
2. modin is installed: Calling `to_snowpark_pandas_dataframe` raises `ImportError: cannot import name '_create_operator_docstring' from partially initialized module 'snowflake.snowpark.modin.utils' (most likely due to a circular import) (/Users/joshi/code/snowpandas/src/snowflake/snowpark/modin/utils.py)`.


**This PR**
As in PR description, the behavior for both cases matches that in release 1.18.

Reply:
I see, if that is the case, this will run under the env where modin is not installed, right? 
Also, i see we have a test suite called test_df_to_snowpark_pandas, the test didn't import snowpark pandas, how come it didn't error out? did the modular got initiated for all integ test? can we have one test that mimic the user's behavior?

Reply:
It seems like I misunderstood how our test environments are set up; you're right that this test runs in the Snowpark Python test environment that does not have modin.

Like you said, the existing test under `tests/integ/modin/test_df_to_snowpark_pandas.py` run with the plugin properly initialized (this happens in `tests/integ/modin/conftest.py`). It looks like if we're to properly test both scenarios we'll need to add a couple new test environments:
(1: no modin installed) Putting the test outside the Snowpark pandas hierarchy (like the `tests/integ/test_df_to_snowpark_pandas.py` file I added here) more or less gives us the desired setup, but the test is currently failing due to a mismatch in pandas version (this environment is using 2.2.2 while modin needs 2.2.1).
(2: modin is installed) This test would have to run outside the Snowpark pandas hierarchy to avoid initializing the plugin, but also needs to be in an environment that has the modin dependency + correct pandas version installed.

Do you have any suggestions for alternative approaches, or should I proceed with adding these extra test environments? Since they'd only be executing 1 or 2 tests, I think it should be OK to add to precommit, but if we want to avoid bloat I can also put them in the daily job instead.

Reply:
I think here is what we can do, we already have test env that have modin installed (snowpark pandas) and env without modin installed (snowpark python), you can right the test to capture behavior on both env, and keep the test under snowpark integration fold, not modin folder. This test will automatically been run under the non-modin env, and then add this test point to snowparkpandasnotdoctest in tox.ini, it will run under the modin env.

For the 2.2.1 and 2.2.2 I felt the modin check should happen first before pandas, because if customer installs modin, it should automatically fix the pandas problem.

Reply:
@sfc-gh-yzou I made the change you suggested (keeping the test in top-level `tests/integ`, and adding this test to tox.ini to run in an environment with modin.

I made the test check for either a `ModuleNotFoundError` (modin not installed) or `RuntimeError` (mismatched pandas version); as in my above comment, we need to check pandas version before modin internals do because modin surfaces a different warning. We can revisit this behavior upstream later.